core/domain/suggestion_services.py

```diff
@@ -599,6 +606,211 @@ def get_translation_suggestions_waiting_longest_for_review_per_lang(
     ]
 
 
+def _get_plain_text_from_html_content_string(html_content_string):
+    """Retrieves the plain text from the given html content string. RTE element
+    occurrences in the html are replaced by its corresponding name capitalized
+    in square brackets.
+    eg: <p>Sample1 <oppia-noninteractive-math></oppia-noninteractive-math>
+        Sample2 </p> will give as output: Sample1 [MATH] Sample2.
+    Note: similar logic exists in the frontend in format-rte-preview.filter.ts.
+
+    Args:
+        html_content_string: str. The content html string to convert to plain
+            text.
+
+    Returns:
+        str. The plain text string from the given html content string.
+    """
+    # Replace frequently used HTML character entities.
+    html_content_string = re.sub('&nbsp;', ' ', html_content_string)
+    html_content_string = re.sub('&quot;', '"', html_content_string)
+    html_content_string = re.sub('&apos;', '\'', html_content_string)
```

Comment:
Can I get reviewers feedback on whether this replacement is necessary? I didn't see &apos; anywhere in the codebase and it's a little awkward to replace it with '\''.

Reply:
Ah -- if the html chars only appear within the oppia-noninteractive-* attributes, then I don't think you need the replacements, since you're going to drop the whole tag anyway. The attr values within those tags are encoded in a special way, so I wouldn't change them ad hoc like you're doing here.

What I meant in my earlier comment was to check whether you can reproduce these other characters outside the context of such tags. If you can, then something is wrong and we should dig deeper into this. But if not, then you don't need the replacement logic.

Reply:
I removed the apostrophe check but what do you think the logic was for replacing the quotes and nbsp in the function in format-rte-preview.filter.ts then? The function in that file is trying to achieve the same behavior as me. I can reach out to who wrote the function and ask. So far, the only character that I've been able to see outside of the oppia-noninteractive-* attributes is the &amp by clicking on the insert code button in the editor...I may have found a bug (see screenshots below).

On the editor UI for submitting a translation I did:
<img width="796" alt="Screen Shot 2020-10-05 at 9 07 18 PM" src="https://user-images.githubusercontent.com/19376937/95158341-37360f80-0750-11eb-8cc3-3ae18795ed6d.png">

The result of the translation that I submitted in the editor on the contributor dashboard homepage:
<img width="1776" alt="Screen Shot 2020-10-05 at 9 15 04 PM" src="https://user-images.githubusercontent.com/19376937/95158294-1a014100-0750-11eb-8174-0d20c236f618.png">

On the console while I'm hovering over the translation that's been submitted:
<img width="621" alt="Screen Shot 2020-10-05 at 9 14 43 PM" src="https://user-images.githubusercontent.com/19376937/95158326-2daca780-0750-11eb-8147-8a0739a2bd47.png">

Reply:
That does look like a bug -- well done on finding it. I think this can be self-investigated, actually: try to construct a piece of html in the frontend with quotation marks as well as the characters "&", "q", "u", etc. one after the other, and see what that function does to it. It's probably going to be faster to just investigate it than to dig into its origin, which based on "git blame" is at least 2 years ago. You can also look for the PR that originally introduced that code to see if the PR comments include any useful context. If you can find it, I'd be happy to help with interpretation.

Also, I think that something is a bit off with the storage of `&&&` as `&amp;&amp;&amp;`. The RTE here should behave the same way as any RTE in the editor -- if you enter `&&&` then it looks like `&&&`  is what is stored, and also what is displayed in the exploration preview. So, this does seem like a legitimate bug with the creator dashboard. Perhaps file it and either investigate or hand off to @sagangwee or @DubeySandeep? (Also /cc @kevintab95 for info.)

Reply:
Yeah I tried to construct something that would trigger creating the character entities yesterday and again today. The only thing that would come up was the &amp; It turns out that you can just put a & as plaintext in the translation submission (you don't have to click any of the rte element buttons) and it will produce the &amp; I will file the bug today.

I used git blame and I think the character entities were replaced before because originally the rte tags weren't converted into [Math] etc. but the replacement code was just kept and not removed. You can see what I mean by looking at PR #4269. As a result, I removed the html character replacements. I think the character replacements could also be removed from the format-rte-preview.filter.ts file as well. I could do that in a separate PR if you'd like.

Reply:
Update - I just realized the & character shows up correctly in the editor for the contributor dashboard page as well. It is just shown incorrectly in the suggestion lists.

@sagangwee @DubeySandeep I filed issue #10889 with the details

Reply:
This seems resolved per the latest implementation, unless I'm missing something.